puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foreman-x-develop-release.groovy

```diff
@@ -15,33 +15,12 @@ pipeline {
     }
 
     stages {
-        stage('Setup Environment') {
+        stage('Copy Source') {
             steps {
                 script {
-                    if ( env.git_ref == null || env.git_ref == '' ) {
-                        error("git_ref parameter is blank")
-                    }
-                }
-                setup_nightly_build_environment(
-                    git_url: git_url,
-                    git_ref: env.git_ref ?: git_ref,
-                    project_name: project_name
-                )
-            }
-        }
-        stage('Archive Git Commit') {
-            steps {
-                script {
-                    dir(project_name) {
-                        commit_hash = archive_git_hash()
-                    }
-                }
-            }
-        }
-        stage('Build and Archive Source') {
-            steps {
-                script {
-                    sourcefile_paths = generate_sourcefiles(project_name: project_name, source_type: source_type)
+                    sourcefile_paths = pwd(tmp: true)
+                    source_project_name = "${project_name}-${git_ref}-source-release"
+                    copyArtifacts(projectName: source_project_name, target: sourcefile_paths)
```

Comment:
Is this here in an attempt to drop the FetchBuilder ?

Reply:
Yes, the current nightlies already don't use them. FetchBuild is still used by scratch builds though

Reply:
> Yes

:+1:

Reply:
For explicitness, I am thinking the originating job should pass its build number and copy artifacts from that build number rather than last successful? If we trust things, then I would opt to explicitly configure to use `lastWithArtifacts` or `lastSuccessful`.

See selector at https://jenkins.io/doc/pipeline/steps/copyartifact/

Reply:
I like that idea and IMHO `lastSuccesful` is the most appropriate, but I struggle a bit with the syntax. `Nested Choice of Objects`, how to specify that?

Reply:
I think:

```
target: sourcefile_paths, selector: {lastSuccessful: true})
```

Reply:
Updated